src/App.tsx

```diff
@@ -15,61 +14,185 @@ const goodsFromServer = [
   'Garlic',
 ];
 
+const minLengths = new Array(10).fill(0).map((_, i) => i + 1);
+
 enum SortType {
   NONE,
-  ALPABET,
+  ALPHABET,
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
-  isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
-
-  // Sort and reverse goods if needed
-  // ...
-
-  return visibleGoods;
-}
-
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
   sortType: SortType,
+  selected: number,
+  isSortedByLength: boolean,
+  isSortedAlphabetically: boolean,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends Component<{}, State> {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+    selected: 10,
+    isSortedByLength: false,
+    isSortedAlphabetically: false,
+  };
+
+  getReorderedGoods = (
+    goods: string[],
+    sortType: SortType,
+    isReversed: boolean,
+    selected: number,
+    isSortedByLength: boolean,
+    isSortedAlphabetically: boolean,
+  ) => {
+    const visibleGoods = goods.filter(good => good.length <= selected);
+
+    // f equal first word and s equal second word
+    visibleGoods.sort((firstWord, secondWord) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return isSortedAlphabetically
+            ? firstWord.localeCompare(secondWord)
+            : secondWord.localeCompare(firstWord);
+        case SortType.LENGTH:
+          return isSortedByLength
+            ? firstWord.length - secondWord.length
+            : secondWord.length - firstWord.length;
+        default:
+          return 0;
+      }
+    });
```

Comment:
Your sort will start every time, check each element and do nothing with him
```suggestion
    switch (sortType) {
      case SortType.ALPABET:
        return visibleGoods.sort((a, b) => a.localeCompare(b));
      case SortType.LENGTH:
        return visibleGoods.sort((a, b) => a.length - b.length);
      case SortType.NONE:
      default:
        break;
    }
```

Reply:
Теперь сортирует один раз, для реверса нужно нажать кнопку реверс